frontend/src/components/SessionTimer.js

```diff
@@ -0,0 +1,224 @@
+import styled from 'styled-components/macro';
+import { useState } from 'react';
+import ButtonBig from './ButtonBig';
+import ButtonTab from './ButtonTab';
+import { GiMeeple } from 'react-icons/all';
+
+export default function SessionTimer({
+    setStartDate,
+    setSessionDuration,
+    session,
+}) {
+    const [timer, setTimer] = useState({ s: 0, m: 0, h: 0 });
+    const [startTimeForSession, setStartTimeForSession] = useState('');
+    const [stopTimeForSession, setStopTimeForSession] = useState('');
+    const [intervalId, setIntervalId] = useState();
+    const [gameHasStarted, setGameHasStarted] = useState(false);
+    const [duration, setDuration] = useState();
+
+    const onClickToStart = () => {
+        let startDate = new Date();
+        let startDateTimeStamp = startDate.getTime();
+        setStartTimeForSession(startDateTimeStamp);
+        setStartDate(startDate);
+        console.log(startDate);
+        console.log(startDateTimeStamp);
+        run();
+        const intervalId = setInterval(run, 1000);
+        setIntervalId(intervalId);
+        setGameHasStarted(true);
+    };
+
+    const onClickToStop = () => {
+        let stopDate = new Date();
+        let stopDateTimeStamp = stopDate.getTime();
+        console.log(stopDate);
+        setStopTimeForSession(stopDateTimeStamp);
+        console.log(stopTimeForSession);
+
+        clearInterval(intervalId);
+        setGameHasStarted(false);
+        const gameDuration = (
+            (new Date().getTime() - startTimeForSession) /
+            1000 /
+            60
+        )
+            .toString()
+            .split('.')[0];
+
+        console.log(gameDuration);
+        setDuration(gameDuration);
+    };
+
+    let updatedS = timer.s,
+        updatedM = timer.m,
+        updatedH = timer.h;
+
+    const run = () => {
+        if (updatedS === 60) {
+            updatedM++;
+            updatedS = 0;
+        }
+        if (updatedM === 60) {
+            updatedH++;
+            updatedM = 0;
+        }
+        updatedS++;
+        return setTimer({ s: updatedS, m: updatedM, h: updatedH });
+    };
+
+    const goToScoring = () => {
+        setSessionDuration(duration);
+    };
+
+    return (
+        <>
+            <Wrapper>
+                <img src={session.imageUrl} alt="game cover" />
+                <section className="timer">
+                    <span>{timer.h >= 10 ? timer.h : '0' + timer.h}</span>
+                    &nbsp;:&nbsp;
```

Comment:
Maybe use padding or margin! HTML is NOT for styling

Reply:
changed